python/cudf/cudf/_lib/hash.pyx

```diff
@@ -51,7 +54,8 @@ def hash_partition(Table source_table, object columns_to_hash,
             else None
 
         ),
-        list(c_result.second)
+        list(c_result.second) if c_result.second.size()
+        else list(vector[libcudf_types.size_type](num_partitions, 0))
```

Comment:
instead of constructing the list from a vector we'd be better off just constructing the list directly. Cython would have to iterate through the vector and convert each int into a python object.

Reply:
Got it - Wasn't sure of the best approach here.  So you are saying a simple `[0] * num_partitions` is sufficient, since we need a list anyway?